projects/cloud/app/services/change_user_role_service.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class ChangeUserRoleService < ApplicationService
+  attr_reader :user_id, :organization_id, :current_role, :new_role, :current_user
+
+  module Error
+    Unauthorized = Class.new(StandardError)
+  end
+
+  def initialize(user_id:, organization_id:, current_role:, new_role:, current_user:)
+    super()
+    @user_id = user_id
+    @organization_id = organization_id
+    @current_role = current_role
+    @new_role = new_role
+    @current_user = current_user
+  end
+
+  def call
+    return user if current_role == new_role
+    ActiveRecord::Base.transaction do
+      organization = Organization.find(organization_id)
+      raise Error::Unauthorized unless OrganizationPolicy.new(current_user, organization).update?
```

Comment:
We need to catch this one at the mutation level and convert that into a standard GraphQL response for errors.